src/main/resources/XWiki/Macros/Team.xml

```diff
@@ -0,0 +1,1407 @@
+#set($configObject = $xwiki.getDocument("XWiki.Macros.TeamConfiguration").getObject("XWiki.Macros.TeamConfigurationClass"))
+#if("$!{size}" == "")
+  #set($size = $configObject.getValue("size"))
+#end
+#if("$!{size}" == "")
+  #set($size = 60)
+#end
+#if("$!{letterAvatarBgColor}" == "")
+  #set($letterAvatarBgColor = $configObject.getValue("letterAvatarBgColor"))
+#end
+#if("$!{letterAvatarBgColor}" == "")
+  #set($letterAvatarBgColor = "#0A6")
+#end
+#if("$!{letterAvatarFontColor}" == "")
+  #set($letterAvatarFontColor = $configObject.getValue("letterAvatarFontColor"))
+#end
+#if("$!{letterAvatarFontColor}" == "")
+  #set($letterAvatarFontColor = "white")
+#end
+#if("$!{requireExternalAuth}" == "")
+  #set($requireExternalAuth = $configObject.getValue("requireExternalAuth"))
+#end
+#if("$!{requireExternalAuth}" == "false" || "$!{requireExternalAuth}" == "0")
+  #set($requireExternalAuth = false)
+#end
+#if("$!{showUsernames}" == "")
+  #set($showUsernames = $configObject.getValue("showUsernames"))
+#end
+#if("$!{showUsernames}" == "false" || "$!{showUsernames}" == "0")
+  #set($showUsernames = false)
+#end
+#if("$!{disableTools}" == "")
+  #set($disableTools = $configObject.getValue("disableTools"))
+#end
+#if("$!{disableTools}" == "false" || "$!{disableTools}" == "0")
+  #set($disableTools = false)
+#end
+#if("$!{disableLetterAvatars}" == "")
+  #set($disableLetterAvatars = $configObject.getValue("disableLetterAvatars"))
+#end
+#if("$!{disableLetterAvatars}" == "false" || "$!{disableLetterAvatars}" == "0")
+  #set($disableLetterAvatars = false)
+#end
+#if(!$tag || $tag == "")
+  #set($tagselect = "")
+  #set($tagwhere = "")
+#else
+  #set($tagselect = ",
+      BaseObject as obj3,
+      DBStringListProperty as tagprop
+    JOIN
+      tagprop.list list
+  ")
+  #set($tagwhere = " and
+    doc.fullName = obj3.name and
+    obj3.className = 'XWiki.TagClass' and
+    obj3.id = tagprop.id.id and
+    tagprop.id.name = 'tags' and
+    list = '${tag}'
+  ")
+#end
+#set($sql = "SELECT doc.fullName FROM XWikiDocument doc,
+    BaseObject as obj
+    #if($requireExternalAuth), BaseObject as obj2 #end,
+    IntegerProperty as prop
+    $tagselect
+  WHERE
+    doc.fullName = obj.name and
+    obj.className = 'XWiki.XWikiUsers'
+    #if($requireExternalAuth) and
+      doc.fullName = obj2.name and (
+        obj2.className = 'XWiki.LDAPProfileClass' or
+        obj2.className = 'XWiki.OIDC.UserClass'
+      )
+    #end and
+    obj.id = prop.id.id and
+    prop.id.name = 'active' and
+    prop.value = 1
+    $tagwhere
+  ORDER BY doc.name
+")
+#set($list = $services.query.hql($sql).execute())
+{{html wiki=false clean=false}}
+&lt;div class="xwikiteam #if($showUsernames)xwikiteam-usernames-displayed #else xwikiteam-usernames-hidden #end"&gt;
+  #if ($list.size() &gt; 0)
+    &lt;ul class="xwikiteam-ul"&gt;
+      #foreach($user in $list)
+        #avatar("xwiki:${user}" $size $disableLetterAvatars $letterAvatarBgColor $letterAvatarFontColor)
```

Comment:
You should **resolve** the local user reference relative to the current wiki (I guess, since the query above is executed in the current wiki, which means the team macro is looking only for users defined in the current wiki), instead of computing the absolute user reference by concatenating the wiki reference (not to mention that this won't work correctly inside a subwiki, that has a different reference than 'xwiki').

Reply:
How can I resolve the local user reference? Is there a way to do this in the hql query directly?

Reply:
Ok, this works without putting "xwiki:". I guess the user will be correctly resolved without.

Reply:
We need to know whether the macro is supposed to list users from the current wiki or from the main wiki. I don't have the answer. I think it makes more sense to list the users from the current wiki (by default at least), but listing the users from the main wiki (even when you are not on the main wiki) is also a valid use case.

For listing the users from the current wiki, removing the "xwiki:" is fine because ``#avatar`` is probably resolving (directly or indirectly) the given user reference relative to the current wiki.

Reply:
That could be a parameter?
I'd suggest we resolve the users in the current wiki and add the parameter to resolve globally when needed/requested?